tests/LSP6KeyManager/tests/LSP6ControlledToken.test.ts

```diff
@@ -0,0 +1,840 @@
+} from "../../../types";
+import {
+  ERC725YKeys,
+  ALL_PERMISSIONS,
+  PERMISSIONS,
+  ERC1271_VALUES,
+} from "../../../constants";
+import { abiCoder, ARRAY_INDEX, ARRAY_LENGTH } from "../../utils/helpers";
+import { BytesLike } from "ethers";
+
+export type LSP6ControlledToken = {
+  accounts: SignerWithAddress[];
+  token: LSP7Mintable | LSP8Mintable;
+  keyManager: LSP6KeyManager;
+};
+
+const buildContext = async () => {
+  const accounts = await ethers.getSigners();
+
+  const lsp7 = await new LSP7Mintable__factory(accounts[0]).deploy(
+    "name",
+    "symbol",
+    accounts[0].address,
+    true
+  );
+
+  const keyManager = await new LSP6KeyManager__factory(accounts[0]).deploy(
+    lsp7.address
+  );
+
+  const keys = [
+    ERC725YKeys.LSP6["AddressPermissions[]"].length,
+    ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.ZERO,
+    ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+      accounts[0].address.substring(2),
+  ];
+
+  const values = [ARRAY_LENGTH.ONE, accounts[0].address, ALL_PERMISSIONS];
+
+  await lsp7.connect(accounts[0])["setData(bytes32[],bytes[])"](keys, values);
+  await lsp7.connect(accounts[0]).transferOwnership(keyManager.address);
+
+  return {
+    accounts,
+    token: lsp7,
+    keyManager,
+  };
+};
+
+const addControllerWithPermission = async (
+  context: LSP6ControlledToken,
+  account: SignerWithAddress,
+  arrayLength,
+  arrayIndex: BytesLike,
+  permissions: BytesLike
+) => {
+  const keys = [
+    ERC725YKeys.LSP6["AddressPermissions[]"].length,
+    ERC725YKeys.LSP6["AddressPermissions[]"].index + arrayIndex,
+    ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+      account.address.substring(2),
+  ];
+
+  const values = [arrayLength, account.address, permissions];
+
+  const payload = context.token.interface.encodeFunctionData(
+    "setData(bytes32[],bytes[])",
+    [keys, values]
+  );
+
+  await context.keyManager.connect(context.accounts[0]).execute(payload);
+};
+
+describe("When deploying LSP7 with LSP6 as owner", () => {
+  let context: LSP6ControlledToken;
+
+  before(async () => {
+    context = await buildContext();
+  });
+
+  it("should have lsp6 as owner of the lsp7", async () => {
+    expect(await context.token.owner()).to.be.equal(context.keyManager.address);
+  });
+
+  it("should set the necessary controller permissions correctly", async () => {
+    const keys = [
+      ERC725YKeys.LSP6["AddressPermissions[]"].length,
+      ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.ZERO,
+      ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+        context.accounts[0].address.substring(2),
+    ];
+
+    const values = [
+      ARRAY_LENGTH.ONE,
+      context.accounts[0].address,
+      ALL_PERMISSIONS,
+    ];
+
+    expect(await context.token["getData(bytes32[])"](keys)).to.be.deep.equal(
+      values
+    );
+  });
+
+  describe("using execute(..) in LSP7 through LSP6", () => {
+    it("should revert", async () => {
```

Comment:
Maybe we could add a bit more descriptions in the tests, so that we know why it should revert?

```suggestion
    it("should revert because function does not exist on LSP7", async () => {
```